engine/model.py

```diff
@@ -51,6 +51,8 @@ class Model:
     scene: Optional[arcade.Scene]
     physics_engine: Optional[arcade.PhysicsEngineSimple]
 
+    time: int
```

Comment:
The word "time" is a bit vague. When people read this there are a few unanswered questions: is this a time of day? A duration? What unit is it in?

A better name might be `millis_passed`.

For educational purposes I'll tell you the "proper" way to do this, but I'm choosing to skip it for performance reasons (and I'm actually still on the fence since you usually shouldn't care too much about performance unless you have a evidence that something is slow). Everything below this don't actually do, but file it away in your memory bank.

The "proper" way would be to use the `timedelta` class provided by Python's standard library:

```python
import datetime

class Model:
  time_passed: datetime.timedelta

  def __init__(self):
    self.time_passed = datetime.timedelta()

  def on_update(self, dt: int):
    self.time_passed += datetime.timedelta(milliseconds=dt)

  @property
  def game_time_passed(self) -> datetime.timedelta:
    return self.time_passed
```
The value of this approach is that you don't have to say what units it is in; `timedelta` represents any arbitrary length of time and can be converted by requesting the proper methods. Most modern languages have this sort of concept somewhere in their standard libraries, simply because there's been so many bugs over the years related to someone accidentally using milliseconds instead of seconds or something like that.